ratecounter.go

```diff
@@ -76,6 +76,18 @@ func (r *RateCounter) Rate() int64 {
 	return r.counter.Value()
 }
 
+func (r *RateCounter) MaxTickRate() int64 {
```

Comment:
Based on circle, seems like theres a race in here. 🤔 

Could be in accessing partials? or maybe Value has something more subtle?